### PR TITLE
Convert ListView and TimelineView to typescript

### PR DESCRIPTION
**Pull request by bwindels**

### CODE REVIEW

File: src/platform/web/ui/general/ListView.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+            this.loadList();
+        }
+    }
+
+    mount(): HTMLElement {
+        const attr: {[name: string]: any} = {};
+        if (this._className) {
+            attr.className = this._className;
+        }
+        this._root = el(this._tagName, attr);
```
Comment (DanilaFe) [inline src/platform/web/ui/general/ListView.ts:82]: What if we do this:
```suggestion
        const root = this._root = el(this._tagName, attr);
```

File: src/platform/web/ui/general/ListView.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        this.loadList();
+        if (this._onItemClick) {
+            this._root!.addEventListener("click", this);
```
Comment (DanilaFe) [inline src/platform/web/ui/general/ListView.ts:85]: Then, this becomes:
```suggestion
            root.addEventListener("click", this);
```

File: src/platform/web/ui/general/ListView.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        }
+        return this._root!;
```
Comment (DanilaFe) [inline src/platform/web/ui/general/ListView.ts:87]: And this,
```suggestion
        return root;
```
In short, no `!` needed?

File: src/platform/web/ui/general/types.ts [hunk 4]
```diff
@@ -13,15 +13,14 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
-    let node;
-    try {
-        node = view.mount(mountArgs);
-    } catch (err) {
-        node = errorToDOM(err);
-    }
-    return node;
-}
\ No newline at end of file
+export interface UIView {
```
Comment (DanilaFe) [inline src/platform/web/ui/general/types.ts:21]: Should we perhaps stick to the convention of starting our interfaces with `I`? it seems to be in use everywhere else.

Comment (bwindels) [reply src/platform/web/ui/general/types.ts:21]: Have done so in https://github.com/vector-im/hydrogen-web/commit/a6bcfac597a22ae3a26f1d528def45a9a0d36497 in #486 as I have made changes there to the usage of the interface.

File: src/platform/web/ui/general/ListView.ts (hunk 3 above)
Comment (bwindels) [reply src/platform/web/ui/general/ListView.ts:87]: Have done so in https://github.com/vector-im/hydrogen-web/commit/b71a26b04a6368921279fd52d26e0a0b21d47e73 in #486

**Pull request merged**